Proposed Dev compass259 changes to Meraki scripts and templates.

compass259 opened the pull request:
Proposed Dev compass259 changes to Meraki scripts and templates.

Itemized changes from original:
- reworked meraki_status.sh script to save every API call to a json file, then append and merge those files together, instead of doing so in memory as variables.  Added the meraki.network.json file generation to this script and renamed it meraki_status_discovery.sh.  This script is meant to be run a couple times a day.  Takes about 2 secs per device.
- Added a meraki_status_quickpoll.sh script for Meraki info that should be recorded often for troubleshooting and historical value.  It contains basic device info, uplink info, loss/latency info, cellular type and signal info.  This script is meant to be run often, such as every 5-10 minutes depending on need.  
- Device Template rework to populate 7 quickpoll info items (uplink type, uplink status, cellular signal type (3G/4G), cellular RSRP, cellular RSRQ, packet less, packet latency) dependent on quickpoll data pull.
- Device template rework to populate 7 device specific items (address, notes, MAC, networkID, serial, tags, device status) dependent on slower device discovery data pull.  The device status check is dependent on the quickpoll data pull so the device status check can be accurately reflected at the set poll interval (5-10 mins).
- network discovery template rework to only include Meraki (network name).
- Included suggested cron scheduling entries.
- Adjusted readme data file to reflect these changes.

zdegraaf committed:
Update meraki_status.sh

Add period to jq command

compass259 committed:
Add files via upload

Updated meraki.sh.  Moved the meraki.network.json generation part of the script into the meraki_status_discovery.sh script for easier cron scheduling.

compass259 committed:
Add files via upload

This script replaces the meraki_status.sh script, adding the meraki.sh meraki.networks.json file generation and other tweaks.

compass259 committed:
Add files via upload

This script is to be run often (ie: every 5 mins) for accurate up to date status of the Meraki network.  Contains base device status, uplink status, cellular signal strength, loss and latency.

compass259 committed:
Add files via upload

Updated Meraki device discovery.  Contains seven items in the uplink discovery, and seven items in the device discovery.

compass259 committed:
Add files via upload

Updated Meraki network discovery template.  Mostly the same as original.

compass259 committed:
Add files via upload

Example of Zabbix cron file for the meraki_status_discovery.sh and meraki_status_quickpoll scripts.

compass259 committed:
Delete meraki_status.sh

compass259 committed:
Delete Zabbix_templates.yaml

compass259 committed:
Update README.md

compass259 committed:
Create README.md

Adjusted working to reflect proposed changes.

zdegraaf reviewed (changes requested):
Can we export both templates under a single yaml file for consistency and simplicity? To export both in one file, check both templates before hitting export.

bgbak commented on Zabbix_Meraki_Network_Discovery_Template.xml:
```diff
@@ -0,0 +1,77 @@
+                            </group_prototypes>
+                            <templates>
+                                <template>
+                                    <name>AL Meraki Device Discovery</name>
```
Incorrect template name for use with the included device discovery template.
This will fail on import.

compass259 committed:
Add files via upload

Combined Meraki templates with new uplink triggers to trigger on packet loss, cellular on 3G, or poor cellular signal strength.

compass259 committed:
Delete Zabbix_Meraki_Device_Discovery_Template.xml

Replacing with combined template.

compass259 committed:
Delete Zabbix_Meraki_Network_Discovery_Template.xml

Replacing with combined template.

compass259 commented:
Deleted both separate templates and re-added combined template file with additional uplink telemetry triggers.

compass259 committed:
Add files via upload

Adjusted the Poor cellular signal trigger to only match -106 to -109 dBm signal as intended.
